src/jit/_typeinfo.h

```diff
@@ -300,8 +304,10 @@ class typeInfo
 
     union {
         CORINFO_CLASS_HANDLE m_cls;
-        // Valid only for type TI_METHOD
+        // Valid only for type TI_METHOD without IsToken
         CORINFO_METHOD_HANDLE m_method;
+        // Valid only for TI_TOKEN with IsToken
+        CORINFO_RESOLVED_TOKEN* m_token;
```

Comment:
You are right. Fixed.
I thought that the global new is overloaded to allocate memory in the memory pool.